src/v/iceberg/tests/schema_protobuf_test.cc

```diff
@@ -0,0 +1,245 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "gtest/gtest.h"
+#include "iceberg/datatypes.h"
+#include "iceberg/schema_protobuf.h"
+#include "iceberg/tests/test_schemas.h"
+#include "test_utils/runfiles.h"
+#include "test_utils/test.h"
+#include "utils/file_io.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <fmt/core.h>
+#include <fmt/ostream.h>
+#include <gmock/gmock.h>
+#include <google/protobuf/compiler/parser.h>
+#include <google/protobuf/descriptor.h>
+#include <google/protobuf/descriptor.pb.h>
+#include <google/protobuf/io/tokenizer.h>
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <memory>
+#include <sstream>
+
+using namespace iceberg;
+using namespace testing;
+namespace {
+static constexpr auto file_descriptor_name = "test_schema_file";
+AssertionResult parse_proto_definition_string(
+  const ss::sstring& proto_str, google::protobuf::DescriptorPool& pool) {
+    google::protobuf::compiler::Parser parser;
+
+    google::protobuf::io::ArrayInputStream input_stream(
+      proto_str.c_str(), proto_str.size());
+
+    google::protobuf::io::Tokenizer tokenizer(&input_stream, nullptr);
+    google::protobuf::FileDescriptorProto fdp;
+    fdp.set_name(file_descriptor_name);
+
+    auto parse_result = parser.Parse(&tokenizer, &fdp);
+    if (!parse_result) {
+        return AssertionFailure(
+          Message("Unable to parse protocol buffer defintion string"));
```

Comment:
FYI you don't have to change but you can also do: `return AssertionFailure() << "Unable to ...";`